components/controls/RadioToggle.tsx

```diff
@@ -18,52 +18,58 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 import * as classNames from 'classnames';
+import * as _ from 'lodash';
 import * as React from 'react';
 import './RadioToggle.css';
 import Tooltip from './Tooltip';
 
-interface Option {
+interface Option<T = string> {
   disabled?: boolean;
   label: string;
   tooltip?: string;
-  value: string;
+  value: T;
 }
 
-interface Props {
+interface Props<T = string> {
   className?: string;
   name: string;
-  onCheck: (value: string) => void;
-  options: Option[];
-  value?: string;
+  onCheck: (value: T | undefined) => void;
+  options: Option<T>[];
+  value?: T;
 }
 
-export default class RadioToggle extends React.PureComponent<Props> {
+export default class RadioToggle<T = string> extends React.PureComponent<Props<T>> {
   static defaultProps = {
     disabled: false,
     value: null
   };
 
   handleChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
     const newValue = e.currentTarget.value;
-    this.props.onCheck(newValue);
+    // Let assum that if we can't found the option we have unselect
+    // RadioToggle
+    const selectedOption = _.find(this.props.options, { label: newValue });
+    this.props.onCheck(selectedOption && selectedOption.value);
   };
 
-  renderOption = (option: Option) => {
+  renderOption = (option: Option<T>) => {
     const checked = option.value === this.props.value;
     const htmlId = this.props.name + '__' + option.value;
     return (
-      <li key={option.value}>
+      <li key={option.label}>
         <input
           checked={checked}
           disabled={option.disabled}
           id={htmlId}
           name={this.props.name}
           onChange={this.handleChange}
           type="radio"
-          value={option.value}
+          value={option.label}
         />
         <Tooltip overlay={option.tooltip || undefined}>
-          <label htmlFor={htmlId}>{option.label}</label>
+          <label title={option.label} htmlFor={htmlId}>
```

Comment:
Should we really have both a tooltip and title ? I'd say in most cases we don't want a title on those `RadioToggle`.
(I'm thinking about how we use them in SC)

Reply:
Indeed it looks like this is too much. Didn't check the result though.